graylog2-server/src/main/java/org/graylog/events/notifications/DBNotificationService.java

```diff
@@ -16,46 +16,77 @@
-import org.graylog2.database.PaginatedDbService;
+import org.graylog2.database.MongoCollections;
 import org.graylog2.database.PaginatedList;
+import org.graylog2.database.pagination.MongoPaginationHelper;
+import org.graylog2.database.utils.MongoUtils;
 import org.graylog2.plugin.database.users.User;
 import org.graylog2.search.SearchQuery;
 
-import jakarta.inject.Inject;
-
+import java.util.Optional;
 import java.util.function.Predicate;
+import java.util.stream.Stream;
 
-public class DBNotificationService extends PaginatedDbService<NotificationDto> {
+import static org.graylog2.database.utils.MongoUtils.insertedIdAsString;
+import static org.graylog2.database.utils.MongoUtils.stream;
+
+public class DBNotificationService {
     private static final String NOTIFICATION_COLLECTION_NAME = "event_notifications";
 
     private final EntityOwnershipService entityOwnerShipService;
+    private final MongoCollection<NotificationDto> collection;
+    private final MongoUtils<NotificationDto> mongoUtils;
+    private final MongoPaginationHelper<NotificationDto> paginationHelper;
 
     @Inject
-    public DBNotificationService(MongoConnection mongoConnection,
-                                 MongoJackObjectMapperProvider mapper,
+    public DBNotificationService(MongoCollections mongoCollections,
                                  EntityOwnershipService entityOwnerShipService) {
-        super(mongoConnection, mapper, NotificationDto.class, NOTIFICATION_COLLECTION_NAME);
+        this.collection = mongoCollections.collection(NOTIFICATION_COLLECTION_NAME, NotificationDto.class);
+        this.mongoUtils = mongoCollections.utils(collection);
+        this.paginationHelper = mongoCollections.paginationHelper(collection);
         this.entityOwnerShipService = entityOwnerShipService;
     }
 
     public PaginatedList<NotificationDto> searchPaginated(SearchQuery query, Predicate<NotificationDto> filter,
                                                           String sortByField, String sortOrder, int page, int perPage) {
-        return findPaginatedWithQueryFilterAndSort(query.toDBQuery(), filter,
-                getSortBuilder(sortOrder, sortByField), page, perPage);
+        return paginationHelper
+                .filter(query.toBson())
+                .sort(sortByField, sortOrder)
+                .perPage(perPage)
+                .postProcessedPage(page, filter);
     }
 
     public NotificationDto saveWithOwnership(NotificationDto notificationDto, User user) {
-        final NotificationDto dto = super.save(notificationDto);
+        final NotificationDto dto = save(notificationDto);
         entityOwnerShipService.registerNewEventNotification(dto.id(), user);
         return dto;
     }
 
-    @Override
+    public NotificationDto save(NotificationDto notificationDto) {
+        if (notificationDto.id() != null) {
+            collection.replaceOne(Filters.eq("_id", new ObjectId(notificationDto.id())), notificationDto);
```

Comment:
I wonder if  we should provide a util for filtering by _id with a string. It's our most common pattern